### PR TITLE
✨ Second step to rename two Kinds in our API

### LINKED ISSUE #1729: feature: Rename two Kinds in our API
### Feature Description

We decided by a [vote](https://kubernetes.slack.com/archives/C058SUSL5AA/p1707155424985459?thread_ts=1706903581.862249&cid=C058SUSL5AA) to rename two Kinds in our API so that
- `Placement` becomes `BindingPolicy`;
- `PlacementDecision` becomes `Binding`.

Right now (immediately after merging of #1681), there are 1728 occurrences of 'placement', including 492 occurrences of 'placementdecisions', case insensitive.

I was suggested to create an issue to specifically track the renaming so here it is.

### Proposed Solution

We have a two-step plan for the renaming.

1. (Small and hard part) Update everything in api/ and test/, plus whatever necessary code changes, to get the new API code available and controllers passing the e2e tests. So that others can begin to contribute their new code growing on the fresh APIs.
2. (Big and easy part) Update everything left: variable names, file/directory names, (go) docs, etc. Also make sure we don't touch things that should not be renamed.

I'll create two PRs, one for each of the two steps, to complete the plan. If there are any (transient) inconsistencies between code and doc in the middle, we should make explicit statements about those inconsistencies.

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by waltforme**
## Summary
This PR takes step 2/2 of https://github.com/kubestellar/kubestellar/issues/1729.

## Related issue(s)

Closes #1729

### CODE REVIEW

File: config/crd/bases/control.kubestellar.io_bindings.yaml [hunk 1]
```diff
@@ -20,9 +20,9 @@ spec:
   - name: v1alpha1
     schema:
       openAPIV3Schema:
-        description: Binding is mapped 1:1 to a single BindingPolicy object. The decision
-          resource reflects the resolution of the BindingPolicy's selectors, and explicitly
-          reflects which objects should go to what destinations.
+        description: Binding is mapped 1:1 to a single BindingPolicy object. It reflects
```
Comment (MikeSpreitzer) [inline config/crd/bases/control.kubestellar.io_bindings.yaml:23]: In English, "it" refers to the most recent noun phrase. In the current revision, that would be that BindingPolicy object, which is not what you are trying to say.

File: api/control/v1alpha1/types.go [hunk 2]
```diff
@@ -169,7 +169,7 @@ type BindingPolicyList struct {
 }
 
 // Binding is mapped 1:1 to a single BindingPolicy object.
-// The decision resource reflects the resolution of the BindingPolicy's selectors,
+// It reflects the resolution of the BindingPolicy's selectors,
```
Comment (MikeSpreitzer) [inline api/control/v1alpha1/types.go:172]: "it" again.

File: config/crd/bases/control.kubestellar.io_bindings.yaml (hunk 1 above)
Comment (waltforme) [reply config/crd/bases/control.kubestellar.io_bindings.yaml:23]: I didn't find an authoritative source where pronoun references are clearly defined. However ChatGPT agrees with my feeling:
image

But I will also respect a native speaker's suggestion and follow up in another PR.

**Pull request merged**